products/llbuild-analyze/Sources/CriticalPathTool.swift

```diff
@@ -6,95 +6,58 @@
-            case criticalPath
-            case all
-        }
-        
-        var database: AbsolutePath!
-        var outputPath: AbsolutePath?
-        var clientSchemaVersion = 9
-        var outputFormat: OutputFormat = .json
-        var graphvizDisplay: GraphvizDisplay = .criticalPath
-        var pretty = false
-        var quiet = false
-        
-        public init() {}
+    enum GraphvizDisplay: String, ExpressibleByArgument {
+        case criticalPath
+        case all
     }
     
-    public override class var toolName: String { return "critical-path" }
+    @Argument(help: "Path to the build database.", transform: { AbsolutePath($0) })
+    var database: AbsolutePath
     
-    public required init(args: [String]) throws {
-        try super.init(
-            usage: "Use this tool to find the critical path in a given build database.",
-            overview: "The tool needs the path to the build database and is able to output the path in multiple different formats.",
-            args: args)
-    }
+    @Option(name: .shortAndLong, help: "Path to generate exported output to.", transform: { AbsolutePath($0) })
+    var output: AbsolutePath?
     
-    override class func defineArguments(parser: ArgumentParser, binder: ArgumentBinder<CriticalPathTool.Options>) {
-        
-        binder.bind(
-            positional: parser.add(positional: "database", usage: "Path to the build database"),
-            to: { $0.database = AbsolutePath($1) })
-        
-        binder.bind(
-            option: parser.add(option: "--output", shortName: "-o", usage: "Optional output path"),
-            to: { $0.outputPath = AbsolutePath($1) })
-        
-        binder.bind(
-            option: parser.add(option: "--clientSchemaVersion", shortName: "-version", usage: "The version of the client schema (default is 9)"),
-            to: { $0.clientSchemaVersion = $1 })
-        
-        binder.bind(
-            option: parser.add(option: "--outputFormat", shortName: "-f", usage: "The format of the output, default is json. Possible values: \(Options.OutputFormat.helpDescription)."),
-            to: { $0.outputFormat = Options.OutputFormat(rawValue: $1) ?? .json })
-        
-        binder.bind(
-            option: parser.add(option: "--pretty", usage: "If output format is json, the output will be pretty printed"),
-            to: { $0.pretty = $1 })
-
-        binder.bind(
-            option: parser.add(option: "--graphvizOutput", usage: "Graphviz display options. Possible values: \(Options.GraphvizDisplay.helpDescription). Default is all"),
-            to: { $0.graphvizDisplay = Options.GraphvizDisplay(rawValue: $1) ?? .all }
-        )
-        
-        binder.bind(
-            option: parser.add(option: "--quiet", shortName: "-q", usage: "There will be no output on stdout."),
-            to: { $0.quiet = $1 })
-        
-    }
+    @Option(default: 9)
```

Comment:
This looks like it needs a couple more parameters:
```suggestion
    @Option(name: [.customLong("clientSchemaVersion"), .customLong("version", withSingleDash: true)], default: 9, help: "The version of the client schema.")
```

Reply:
Thanks, that makes it 100% compatible. I don't expect infrastructure to rely on this yet but it's still better to keep it compatible.